Adding parallel_dd test [v4]

poojabs7 opened the pull request:
This test measures the performance of writing and reading multiple streams of files onto the files system.

Signed-off-by: poojabs pooja@oc2013156016.ibm.com

narasimhan-v commented on io/disk/parallel_dd.py:
```diff
@@ -0,0 +1,231 @@
+        for option in self.dd_roptions.split():
+            cmd += " %s=%s" % (option.split(":")[0], option.split(":")[1])
+        process.run(cmd + ' > /dev/null', shell=True)
+
```
Can you do something like this, to overload the function for read and write..

    def raw_io(self, operation=''):
        """
        Performs raw io operation.
        """
        self.log.info("Timing raw %s of %d megabytes", % (operation, self.megabytes))
        cmd = 'dd '
        if operation == 'read':
            cmd += 'if=%s of=/dev/zero ' % self.fsys.device
            option = self.dd_roptions
        if operation == 'write':
            option = self.dd_woptions
            cmd += 'if=/dev/zero of=%s ' % self.fsys.device
        cmd += 'bs=4k count=%d' % self.blocks

        for option in options.split():
            cmd += " %s=%s" % (option.split(":")[0], option.split(":")[1])
        process.run(cmd, shell=True)

Note: untested code

poojabs7 commented:
Link for V2
https://github.com/avocado-framework/avocado-misc-tests/pull/87/
Change log:
Added README file.
Removed default value for  disk parameter from yaml and added error message in the code.

narasimhan-v commented on io/disk/parallel_dd.py:
```diff
@@ -0,0 +1,231 @@
+        start = time.time()
+        self.fs_read()
+        self.fs_read_rate = self.megabytes / (time.time() - start)
+
```
Since you are doing the same thing multiple times here, can you optimise this code ?

PraveenPenguin commented on io/disk/parallel_dd.py:
```diff
@@ -0,0 +1,231 @@
+            self.blocks_per_file = self.blocks / self.streams
+
+        root_fs_device_cmd = "df | egrep /$ | awk '{print $1}'"
+        root_fs_device = process.system_output(root_fs_device_cmd, shell=True)
```
can `root_fs_device = process.system_output('df | egrep /$ | awk '{print $1}', shell=True)` just want to avoid extra variable ?

PraveenPenguin commented on io/disk/parallel_dd.py:
```diff
@@ -0,0 +1,231 @@
+            device = self.fsys.device
+        try:
+            line = process.system_output('egrep ^%s %s' % (device, s_file))
+            logging.debug(line)
```
why` logging.debug`  i think you can use `self.log.info `as you used earlier ?

PraveenPenguin commented on io/disk/parallel_dd.py.data/parallel_dd.yaml:
```diff
@@ -0,0 +1,24 @@
+    dd_roptions:
+    fs_dd_woptions:
+    fs_dd_roptions:
+
```
please  remove this white space

PraveenPenguin commented:
@poojabs7 ping

poojabs7 commented:
Raised New PR 

https://github.com/avocado-framework-tests/avocado-misc-tests/pull/188
